common/src/main/scala/org/apache/comet/CometConf.scala

```diff
@@ -467,6 +467,15 @@ object CometConf extends ShimCometConf {
       .booleanConf
       .createWithDefault(false)
 
+  val COMET_EXEC_MEMORY_POOL_TYPE: ConfigEntry[String] = conf("spark.comet.exec.memoryPool")
+    .doc(
+      "The type of memory pool to be used for Comet native execution. " +
+        "Available memory pool types are 'greedy', 'fair_spill', 'greedy_task_shared', " +
+        "'fair_spill_task_shared', 'greedy_global' and 'fair_spill_global', By default, " +
+        "this config is 'greedy_task_shared'.")
```

Comment:
We should add that this config only takes effect when off-heap memory is disabled.

Reply:
We can do that in a follow up PR though, and add some more detailed documentation in the tuning guide.

Reply:
Yea, I am not sure if all users can understand the difference between these pool configs. It is better we can add more description.